### PR TITLE
Follow up PR of #2955

### PR DESCRIPTION
**Pull request by mjkkirschner**
continues: 
https://github.com/DynamoDS/DynamoRevit/pull/2955

removes the dynamic call, updates the nugets to use the new flag api.
Also removes the package ref on ionic zip, we stopped relying on it a long time a ago.

### CODE REVIEW

File: src/Config/packages-template.aget [hunk 1]
```diff
@@ -13,8 +13,7 @@
       "Newtonsoft.Json": "8.0.3",
       "NUnit": "2.6.3",
       "Prism": "4.1.0",
-      "RestSharp": "106.12.0",
-      "Unofficial.Ionic.Zip": "1.9.1.8"
```
Comment (QilongTang) [inline src/Config/packages-template.aget:17]: @Mikhinja Please confirm it is fine to remove such dependency as part of DynamoRevit 2.19 release

**Review by QilongTang — APPROVED** [review]

File: src/Config/packages-template.aget (hunk 1 above)
Comment (mjkkirschner) [reply src/Config/packages-template.aget:17]: @QilongTang I checked the DynamoRevit addin and do not see this binary there.

Comment (mjkkirschner) [reply src/Config/packages-template.aget:17]: Also note that this PR is to the master branch - which I assume will not be subject to such reasons for 2 reasons.
1. potential 3.0
2. GL25 is supposed to resolve patch installer insanity.

Comment (Mikhinja) [reply src/Config/packages-template.aget:17]: Doesn't look like it's actually used in DynamoRevit nodes, not sure why it's there.

**Review by Mikhinja — APPROVED** [review]

File: src/DynamoRevit/DynamoRevit.cs [hunk 2]
```diff
@@ -1100,9 +1100,7 @@ private static void OnSplashScreenClosed(object sender, EventArgs e)
             view.Closed -= OnSplashScreenClosed;
             //if the user explicitly closed the splashscreen, then we should let them boot
             //dynamo back up.
-            //TODO use new api by updating dynamocorewpf nuget.
-            //After it's published, remove the use of dynamic.
-            if(sender is Dynamo.UI.Views.SplashScreen ss && ((dynamic)ss).CloseWasExplicit)
+            if(sender is Dynamo.UI.Views.SplashScreen ss && ss.CloseWasExplicit)
```
Comment (Mikhinja) [inline src/DynamoRevit/DynamoRevit.cs:1103]: Do we need these changes in Revit 2024.2 as well?

File: src/Config/packages-template.aget (hunk 1 above)
Comment (QilongTang) [reply src/Config/packages-template.aget:17]: hmm @mjkkirschner I thought this change is for R2024.2?

File: src/DynamoRevit/DynamoRevit.cs (hunk 2 above)
Comment (mjkkirschner) [reply src/DynamoRevit/DynamoRevit.cs:1103]: yes, or users will be able to open multiple instances of Dynamo.

**Pull request merged**